src/main.py

```diff
@@ -91,3 +122,73 @@ def send_error(self, category: str, error_msg: str, **kwargs):
     def send_stack_trace(self, category: str, stack_trace: str, **kwargs):
         if self.consent:
             self.sender.send(self.backend, self.backend.build_stack_trace_message(category, stack_trace, **kwargs))
+
+    @staticmethod
+    def _update_opt_in_status(tid: str, new_opt_in_status: bool):
+        """
+        Updates opt-in status.
+        :param tid: ID of telemetry base.
+        :param new_opt_in_status: new opt-in status.
+        :return: None
+        """
+        app_name = 'opt_in_out'
+        app_version = '1.0.0'
+        opt_in_checker = OptInChecker()
+        opt_in_check = opt_in_checker.check()
```

Comment:
it's better to simplify the logic below to use `prev_status` variable
```
prev_status = OptInStatus.UNDEFINED
if op_in_check == ISIPCheckResult.DECLINED
    prev_status = OptInStatus.DECLINED
elif op_in_check == ISIPCheckResult.ACCEPTED
    prev_status = OptInStatus.ACCEPTED
```

I think it is much more readable

Reply:
Done.